client/selector/default.go

```diff
@@ -46,16 +46,22 @@ func (c *registrySelector) Select(service string, opts ...SelectOption) (Next, e
 		opt(&sopts)
 	}
 
-	// get the service
-	// try the cache first
-	// if that fails go directly to the registry
-	services, err := c.rc.GetService(service)
-	if err != nil {
-		if err == registry.ErrNotFound {
-			return nil, ErrNotFound
-		}
+	// get the service. Because the service could be running in the current or the default domain,
+	// we call both. For example, go.micro.service.foo could be running in the services current domain,
+	// however the runtime (go.micro.runtime) will always be run in the default domain.
+	services, err := c.rc.GetService(service, registry.GetDomain(c.so.Domain))
```

Comment:
Is services always a non nil slice?

Reply:
I believe so, however I'll add a check. This code is being removed soon.